[Natania Yovela] iP

nataniayp opened the pull request:
# Duke
As Nat said:
> Who has time to use other apps for to-do lists?

Introducing Duke, Nat's personalised chat bot, adapted from the Duke project. 🤓

* text-based
* ~~easy~~ *SUPER EASY* to use
* fuss-free
* **$0!** 🤑

## Features
- [x] add task as todo, deadline or event
- [x] delete task
- [x] mark task as done
- [ ] sort tasks based on deadline (coming soon!)
- [ ] reminders (coming soon!)

## How to use
1. Run `Duke.main()` It will trigger this method:

```java
public static void main(String[] args) {
  new Duke("data/duke.txt").run();
}
```

2. Enjoy the bot!

What are you waiting for? Download it here! [click me](https://github.com/nataniayp/ip)

Commit:
Add Gradle support

nataniayp committed:
added greet, echo, exit

nataniayp committed:
added add, list

nataniayp committed:
added mark as done

nataniayp committed:
edited lvl 3, use functions instead

nataniayp committed:
added toString to task, edited printing format

nataniayp committed:
added todos, events, deadlines

nataniayp committed:
added exceptions

nataniayp committed:
added i/o testing, put scanner outside while loop in main

nataniayp committed:
added delete

nataniayp committed:
added delete

nataniayp committed:
merged, fixed conflicts

nataniayp committed:
add javadoc and edit i/o testing

nataniayp committed:
create file in main, add filewriter in addTask to data/duke.txt

nataniayp committed:
add complete feature of saving task to harddisk, add more exceptions to markAsDone and deleteTask

nataniayp committed:
add LocalDate and LocalTime

nataniayp committed:
Merge branch 'branch-Level-7'

nataniayp committed:
Merge branch 'branch-Level-8'

nataniayp committed:
add more oop concepts

nataniayp committed:
divide classes into packages

nataniayp committed:
add JUnit tests

nataniayp committed:
add more JavaDoc comments

nataniayp committed:
Change nothing

nataniayp committed:
add find feature

nataniayp committed:
Merge branch 'branch-A-JavaDoc'

nataniayp committed:
Merge tag 'A-CodingStandard'

nataniayp committed:
Merge tag 'Level-9'

nataniayp committed:
refactor tasklist to tasks

nataniayp committed:
Update README.md

nataniayp committed:
add gradle wrapper files

nataniayp committed:
Merge branch 'branch-A-Gradle'

nataniayp committed:
Merge branch 'add-gradle-support'

nataniayp committed:
merge add-gradle-support to master

nataniayp committed:
use checkstyle to detect coding style violations

nataniayp committed:
Merge branch 'branch-A-Checkstyle'

ooawagaeri commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,93 @@
+     * @return A Map containing the type, description and time (if applicable) of a Task.
+     * @throws DukeException if the input is in invalid format.
+     */
+    public static Map<String, String> parseTextFromInput(String input) throws DukeException {
```
Possible to simplify? parseFromInput?

ooawagaeri commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,93 @@
+     * @param input The text from the file containing the task list.
+     * @return A Map containing the type, description and time (if applicable) of a Task.
+     */
+    public static Map<String, String> parseTextFromFile(String input) {
```
Possible to simplify? parseFromFile?

ooawagaeri reviewed (commented):
Good day Natania, 

Honestly a good job for majority of your code quality (naming).
Just a couple of comments for suggestions.

Good luck for the rest of you iP.

nataniayp committed:
Add GUI

nataniayp committed:
Merge branch 'branch-Level-10'

pcgiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,236 @@
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
```
You may want to update this description!

pcgiang commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,157 @@
+        String s = "";
+        int len = tasks.size();
+        if (len == 0) {
+//            System.out.println("The list is empty!");
```
You may want to delete commented code 😄

pcgiang commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,93 @@
+                String date = LocalDate.parse(time.substring(0, 11),
+                        DateTimeFormatter.ofPattern("dd MMM yyyy")).toString();
+                time = date + " " + time.substring(12);
+            } else if (time.contains(":")) {
```
Is this else if block needed?

pcgiang reviewed (commented):
Hi Nat :)) Overall the code is well structured and very easy to follow! Naming convention and javadocs look great too!

nataniayp committed:
Use assert feature

There are not many things to assert, except for some string returns that
should not be empty. Otherwise, other codes are straightforward and lead
to direct changes that would make assert redundant to add.

nataniayp committed:
Improve code quality

This is done by renaming some methods to so the purpose is better
understood through the names.

nataniayp committed:
Add skeleton for commands

I create a package for commands so that the code is better structured.
The classes are not functional and hence have not been integrated to the
codebase.

nataniayp committed:
Fully integrate command classes

This is done by erasing methods that are independent of command classes,
such that when user input is parsed, it will return a command that can
be executed.

nataniayp committed:
Merge branch 'branch-A-Assertions'

nataniayp committed:
Fix conflicts and merge branch-A-CodeQuality

nataniayp committed:
Merge pull request #1 from nataniayp/branch-A-CodeQuality

Branch a code quality

nataniayp committed:
Add CI:

nataniayp committed:
Delete IO redirection test

nataniayp committed:
Add skeleton of UndoCommand

nataniayp committed:
Add Undo command

nataniayp committed:
Remove System.out

nataniayp committed:
Pull from origin and merge conflicts

nataniayp committed:
Merge pull request #2 from nataniayp/branch-A-CI

Add CI

nataniayp committed:
Merge pull request #3 from nataniayp/branch-C-Undo

Branch c undo

nataniayp committed:
Update README.md

nataniayp committed:
Update README.md

nataniayp committed:
Set theme jekyll-theme-minimal

nataniayp committed:
Update README.md

nataniayp committed:
Fix GUI, add background, margin and padding

nataniayp committed:
Update User Guide

nataniayp committed:
Add Ui.png in docs folder

nataniayp committed:
Merge branch 'master' of https://github.com/nataniayp/ip

nataniayp committed:
Add more JUnit tests
